### PR TITLE
feat(runtime): add stat and statSync methods to Deno.File

### LINKED ISSUE #10109: Add Deno.File.stat and Deno.File.statSync methods
Continuing from #10106

We should add stat and statSync to the Deno.File class.

### PR DESCRIPTION
**Pull request by caspervonb**
Closes #10109

### CODE REVIEW

File: runtime/js/30_files.js
Comment (caspervonb) [inline runtime/js/30_files.js:113]: Tests incoming

**Comment by ry** [comment]
SGTM - any one have objections?

**Comment by kitsonk** [comment]
> SGTM - anyone have objections?

Nope.  But I wanted another opinion other than my own.

**Comment by caspervonb** [comment]
Blocked on #10108

Should be merged first, since this adds directly to the stable File definition and the underlying ops are currently unstable.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**
